biz.aQute.bndlib/src/aQute/bnd/build/WorkspaceExtensionHandler.java

```diff
@@ -0,0 +1,164 @@
+				.error("Could not update avaialbe WorkspaceExtensions", e);
+			return false;
+		}
+		if (extensions != null) {
+			extensions = processor.getReplacer()
+				.process(extensions);
+			Parameters extensionsParameters = new Parameters(extensions, processor);
+			for (Entry<String, Attrs> entry : extensionsParameters.entrySet()) {
+				String extensionName = Processor.removeDuplicateMarker(entry.getKey());
+				String versionString = entry.getValue()
+					.get(WorkspaceExtensionNamespace.CAPABILITY_VERSION_ATTRIBUTE);
+
+				try {
+
+					processor.getLogger()
+					.info("Adding workspace extension {} in version {}", extensionName, versionString);
+
+					Result<Capability> capabilityResult = WorkspaceExtensionNamespace
+						.findWorkspaceExtension(extensionName, versionString, availableWorkspaceExtensions);
+
+					if (capabilityResult.isErr()) {
+						processor.error(capabilityResult.error()
+							.get());
+						continue;
+					}
+					Capability cap = capabilityResult.unwrap();
+
+					// if it is already enabled everything is fine and we need
+					// to
+					// stop
+					if (enabledWorkspaceExtensions.containsKey(cap)) {
+						continue;
+					}
+					org.osgi.resource.Resource resource = cap.getResource();
+					String include = (String) cap.getAttributes()
+						.get(WorkspaceExtensionNamespace.CAPABILITY_INCLUDE_ATTRIBUTE);
+
+					if (include == null) {
+						processor.error(
+							"Can't load workspace extension %s with version %s, because the where directive is missing.",
+							extensionName, versionString);
+						continue;
+					}
+
+					Result<File> result = ws.getBundle(resource, ResourceRepositoryStrategy.REPOS);
+					if (result.isErr()) {
+						processor.error("Can't load workspace extension %s with version %s. Cause: %s", extensionName,
+							versionString, result.error()
+								.get());
+						continue;
+					}
+
+					try (Jar bundle = new Jar(result.unwrap())) {
+						File cacheFolder = ws
+							.getCache("workspaceextensions/" + extensionName + "/" + bundle.getVersion());
+						Version jarVersion = new Version(bundle.getVersion());
+						if (jarVersion.isSnapshot() && cacheFolder.exists()) {
+							cacheFolder.delete();
+						}
+						bundle.writeFolder(cacheFolder);
+
+						File includeFile = IO.getFile(cacheFolder, include);
+						if (includeFile == null || !includeFile.exists()) {
+							processor.error(
+								"Workspace Extension %s with Version %s seems corrupt. The stated file to include %s can't be found.",
+								extensionName, versionString, include);
+							continue;
+						}
+						processor.doIncludeFile(includeFile, false, processor.getProperties(), extensionName);
+						enabledWorkspaceExtensions.put(cap, cacheFolder);
+						doneSomething = true;
+					}
+				} catch (Exception e) {
+					processor.getLogger()
+						.error("Could not apply workspace extension " + extensionName + " in version " + versionString,
+							e);
+				}
+			}
+		}
+		processor.clearPlugins();
```

Comment:
I think you need to call a different method on Processor to clean the plugins. Like `propertiesChanged`.

Reply:
`propertiesChanged` might be a good idea, because this will solve another problem I had. I have to call it at least once though, because my mechanism initializes the plugins in a relatively early stage. This caused a couple of tests to fail. So, we will to the properties changed at least twice for each project/workspace initialization.